quarkus-extension/engine/deployment/src/test/java/org/camunda/bpm/quarkus/engine/test/config/CamundaEngineJobExecutorConfigTest.java

```diff
@@ -44,22 +44,22 @@ public void shouldLoadJobExecutorThreadPoolProperties() {
     // given a custom application.properties file
 
     // then
-    assertThat(config.jobExecutor.threadPool.maxPoolSize).isEqualTo(12);
-    assertThat(config.jobExecutor.threadPool.queueSize).isEqualTo(5);
+    assertThat(config.jobExecutor().threadPool().maxPoolSize()).isEqualTo(12);
+    assertThat(config.jobExecutor().threadPool().queueSize()).isEqualTo(5);
   }
 
   @Test
   public void shouldLoadJobAcquisitionProperties() {
     // given a custom application.properties file
 
     // then
-    assertThat(config.jobExecutor.genericConfig.get("max-jobs-per-acquisition")).isEqualTo("5");
-    assertThat(config.jobExecutor.genericConfig.get("lock-time-in-millis")).isEqualTo("500000");
-    assertThat(config.jobExecutor.genericConfig.get("wait-time-in-millis")).isEqualTo("7000");
-    assertThat(config.jobExecutor.genericConfig.get("max-wait")).isEqualTo("65000");
-    assertThat(config.jobExecutor.genericConfig.get("backoff-time-in-millis")).isEqualTo("5");
-    assertThat(config.jobExecutor.genericConfig.get("max-backoff")).isEqualTo("5");
-    assertThat(config.jobExecutor.genericConfig.get("backoff-decrease-threshold")).isEqualTo("120");
-    assertThat(config.jobExecutor.genericConfig.get("wait-increase-factor")).isEqualTo("3");
+    assertThat(config.jobExecutor().genericConfig().get("max-jobs-per-acquisition")).isEqualTo("5");
+    assertThat(config.jobExecutor().genericConfig().get("lock-time-in-millis")).isEqualTo("500000");
+    assertThat(config.jobExecutor().genericConfig().get("wait-time-in-millis")).isEqualTo("7000");
+    assertThat(config.jobExecutor().genericConfig().get("max-wait")).isEqualTo("65000");
+    assertThat(config.jobExecutor().genericConfig().get("backoff-time-in-millis")).isEqualTo("5");
+    assertThat(config.jobExecutor().genericConfig().get("max-backoff")).isEqualTo("5");
+    assertThat(config.jobExecutor().genericConfig().get("backoff-decrease-threshold")).isEqualTo("120");
+    assertThat(config.jobExecutor().genericConfig().get("wait-increase-factor")).isEqualTo("3");
```

Comment:
❓ Do you know if it makes a difference in performance to only call `config.jobExecutor().genericConfig()` once? It won't have a great effect anyway, but what would be the general best practice?

```suggestion
    Map<String, String> genericConfig = config.jobExecutor().genericConfig();
    assertThat(genericConfig.get("max-jobs-per-acquisition")).isEqualTo("5");
    assertThat(genericConfig.get("lock-time-in-millis")).isEqualTo("500000");
    assertThat(genericConfig.get("wait-time-in-millis")).isEqualTo("7000");
    assertThat(genericConfig.get("max-wait")).isEqualTo("65000");
    assertThat(genericConfig.get("backoff-time-in-millis")).isEqualTo("5");
    assertThat(genericConfig.get("max-backoff")).isEqualTo("5");
    assertThat(genericConfig.get("backoff-decrease-threshold")).isEqualTo("120");
    assertThat(genericConfig.get("wait-increase-factor")).isEqualTo("3");
```

Reply:
- The JVM might perform optimisations under the hood either way so i'm not sure about a real performance benefit
- As a best practice, favouring readability and reusability the chained call could be performed one time instead of multiple.

Personally, i'm not negative about these changes but i don't see them as part of this change. Considering also that they are tests and we change that snippet, it would be nice to do that across all tests, i would personally vote to skip this refactoring or extract it in another context.